cmd/config/internal/commands/cmdcreatesetter.go

```diff
@@ -138,6 +148,51 @@ func (r *CreateSetterRunner) preRunE(c *cobra.Command, args []string) error {
 	return nil
 }
 
+func (r *CreateSetterRunner) processSchema() error {
+	sc, err := schemaFromFile(r.SchemaPath)
+	if err != nil {
+		return err
+	}
+
+	flagType := r.CreateSetter.Type
+	var schemaType string
+	switch {
+	// json schema allows more than one type to be specified, but openapi
+	// only allows one. So we follow the openapi convention.
+	case len(sc.Type) > 1:
+		return errors.Errorf("only one type is supported: %s",
+			strings.Join(sc.Type, ", "))
+	case len(sc.Type) == 1:
+		schemaType = sc.Type[0]
+	}
+
+	// Since type can be set both through the schema file and through the
+	// --type flag, we make sure the same value is set in both places. If they
+	// are both set with different values, we return an error.
+	switch {
+	case flagType == "" && schemaType != "":
+		r.CreateSetter.Type = schemaType
+	case flagType != "" && schemaType == "":
+		sc.Type = []string{flagType}
+	case flagType != "" && schemaType != "":
+		if flagType != schemaType {
+			return errors.Errorf("type provided in type flag (%s) and in schema (%s) doesn't match",
+				r.CreateSetter.Type, sc.Type[0])
+		}
+	}
+
+	// Only marshal the properties in SchemaProps. This means any fields in
+	// the schema file that isn't recognized will be dropped.
+	//TODO: Consider if we should return an error here instead of just dropping
```

Comment:
Nit: space after //

Reply:
Just a thought, can we throw a generic warning message : "processing input schema, any fields which are not valid openAPI will be dropped" while we figure out better UX ?

Reply:
Fixed the missing space.
I don't think we should print a warning message every time, it just makes the experience more confusing. I'll try to find time to actually detect when we do drop any fields and we can return error/warning at that point. I prefer to do that in a separate PR.